ls1intum/Artemis
# `Communication`: Fix several issues and add bookmarking of conversations

### Checklist
#### General


#### Server
#### Client

### Changes | Bugfixes

Bug 1: Non-gender neutral language in English translation. I resolved this issue by changing the translation, as seen in commit dc0bfb0f396b26a6fd0d4d7369de808396db0fe9.

Bug 2: The look of the properties in the info tab of a conversation (see screenshot), were not liked. Adapted the look in 097b46b35e4b054fc3fc99692dbb334b7112dad0

Bug 3: Long channel descriptions and topics caused a scrollbar in the property textbox. Increased the size in commit e1b96ae6d378e03614c560726cf2036f049ed581

Bug 4: Long channel descriptions would cause overflow in channel overview. Fixed in 9c24915eccc6a30bd2c48041110f1c21da89df25

Bug 5: For the conversation filter in the conversation sidebar the magnifying glass was found to be confusing. I adapted this to use a filter icon in : e7fb4f693b64032af458458251f3b884b40e33f0

Bug 6: The new tag shown in the question and answer system. I made sure it shows only for messages in 8c75639031b41a473fb1c68e2a1a1d05ad5847e3

Bug 7: When opening the public channel overview modal, then joining a channel and immediately clicking on the 'View' icon to switch to that newly joined channel, the messages of the previous active conversation are still loaded. I fixed this in 8cebaa13baa6c7f5e183982647f799ba6aa8ed29

😍 **New Feature:** When you open a conversation, it will be stored in the URL and can be bookmarked. When you open the bookmark, the respective conversation will be opened. I added this in 18a189b775528cd1c7a587b683792bc0ff0fcac7



### Steps for Testing

Test Case 1: Verify that gender neutral language is used in English translations.

Test Case 2: Verify that the properties in the info tab of a conversation have a good look.

Test Case 3: Verify that long channel descriptions and topics no longer cause a scrollbar in the property textbox. 250 chars mak

Test Case 4: Verify that long channel descriptions no longer cause overflow in the channel overview. 250 chars mak

Test Case 5: Verify that the magnifying glass in the conversation filter in the conversation sidebar has been replaced with a filter icon.

Test Case 6: Verify that the new tag only shows for messages in the messages system and not for question and answers.

Test Case 7: Verify that when opening the public channel overview modal, joining a channel, and immediately clicking on the 'View' icon to switch to that newly joined channel, the messages of the previous active conversation are no longer loaded but instead of the new conversation.

Test Case 8: Verify that when a conversation is opened, it is stored in the URL and can be bookmarked. Verify that when the bookmark is opened, the respective conversation is opened.








### Review Progress


#### Code Review
#### Manual Tests

### Test Coverage
- No changes

### Screenshots
![new_property_look](https://user-images.githubusercontent.com/29718932/212924567-c253bd4a-9ab4-4298-8efb-7fb72befae6e.png)
![image](https://user-images.githubusercontent.com/29718932/212924633-f4aabbc1-1915-4587-863c-49d12dc5b0db.png)
